app/models/endpoint.rb

```diff
@@ -6,7 +6,7 @@ class Endpoint < ApplicationRecord
   default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
   validates :verify_ssl, :inclusion => {:in => [OpenSSL::SSL::VERIFY_NONE, OpenSSL::SSL::VERIFY_PEER]}
   validates :port, :numericality => {:only_integer => true, :allow_nil => true, :greater_than => 0}
-  validates :url, :uniqueness => true, :if => :url
+  validates :url, :uniqueness => true, :allow_blank => true, :if => :should_have_unique_url?
```

Comment:
How about `:allow_blank => true, :unless => resource.try(:allow_duplicate_url?)`

Reply:
I don't don't think it's good, because:
1) `allow_duplicate_url` method name in `resource` says nothing about which url we mean.
2) `resource.try` in your example will be executed on class initialization. it's possible to put it in lambda, but what the difference with method I made already then?

Reply:
As a project we do prefer `unless` over `if !`, rubocop didn't catch this because it is split over two methods but the concept remains the same

Reply:
This way we can have similar names for the methods e.g. `allow_duplicate_url?` here and `allow_duplicate_endpoint_url?` on `ExtManagementSystem`.

Reply:
please note, that `resource` isn't just an `ExtManagementSystem` - it's polymorphic, which means, we have no guarantee about that method existence.
but ok.

Reply:
Yes which is why having the `.try()` is still a good idea, but I still prefer obvious to implicit and currently ExtManagementSystem is the only class that has endpoints.

Reply:
done